main.py

```diff
@@ -0,0 +1,113 @@
+import os
+
+from flask import Flask, request, abort, send_file
+from google.cloud.exceptions import NotFound
+
+from gaepypi import Package, GCStorage, GAEPyPIError, PackageIndex
+from gaepypi._decorators import basic_auth
+from google.appengine.api import app_identity, wrap_wsgi_app
+
+app = Flask(__name__)
+app.wsgi_app = wrap_wsgi_app(app.wsgi_app)
+storage = None
+
+
+def get_storage():
+	global storage
+	if not storage:
+		bucket_name = os.environ.get('BUCKET_NAME')
+		if not bucket_name:
+			bucket_name = app_identity.get_default_gcs_bucket_name()
+		storage = GCStorage(bucket_name)
+	return storage
+
+
+@app.route("/")
+@basic_auth()
+def root():
+	return '<a href="/packages">packages</a>'
+
+
+@app.route("/pypi/")
+@basic_auth()
+def root_pypi():
+	return get_storage().to_html(full_index=True)
+
+
+@app.route("/", methods=['POST'])
+@basic_auth(required_roles=['write'])
+def root_post():
+	name = request.values.get('name')
+	version = request.values.get('version')
+	action = request.values.get(':action')
+
+	if name and version and len(request.files) and action == 'file_upload':
+		try:
```

Comment:
I always prefer to put try/except closer to the potential issue, so in this case I'd put it before `package = Package(get_storage(), name, version)`